### PR TITLE
hostedcluster: refactor delete

### PR DESCRIPTION
**Pull request by sjenning**
@ironcladlou includes changed from https://github.com/openshift/hypershift/pull/44 and does away with `ensureClusterDeleted`

### CODE REVIEW

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 1]
```diff
@@ -98,13 +98,13 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
...
-		result, err := r.delete(ctx, req)
+		completed, err := r.delete(ctx, hcluster)
 		if err != nil {
 			r.Log.Error(err, "failed to delete cluster", "cluster", req.NamespacedName)
 			return ctrl.Result{}, err
 		}
-		if result != nil {
-			return *result, err
+		if !completed {
+			return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
```
Comment (enxebre) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:107]: nit: might want to put 5 into a constant

File: hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go [hunk 2]
```diff
@@ -491,61 +492,47 @@ func (r *HostedClusterReconciler) listNodePools(clusterNamespace, clusterName st
...
 		}
 	}
 
-	r.Log.Info("Deleting Cluster", "clusterName", req.Name, "clusterNamespace", targetNamespace)
+	r.Log.Info("Deleting Cluster", "clusterName", hc.Name, "clusterNamespace", hc.Name)
 	cluster := &capiv1.Cluster{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      req.Name,
-			Namespace: targetNamespace,
+			Name:      hc.Name,
```
Comment (enxebre) [inline hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:513]: subjective nit: having a variable `targetNamespace` storing `hc.Name` and use it in these references makes it more readable an intuitive to me. Also easier to not get mistaken if more need to be added.

Comment (ironcladlou) [reply hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:513]: The manifests.HostedControlPlaneNamespace struct encapsulates the namespace resource's details; we shouldn't be leaking out the naming strategy anywhere, and could instead do something like:

```
targetNamespace := manifests.HostedControlPlaneNamespace{HostedCluster: hcluster}.Build()
capiCluster := controlplaneoperator.CAPICluster{...}.Build()
```

Comment (ironcladlou) [reply hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go:513]: Let's fix up the hard-coded naming assumptions separately

**Comment by ironcladlou** [comment]
/lgtm
/hold

Feel free to remove the hold when you're done testing!

**Comment by sjenning** [comment]
@ironcladlou found one issue during testing.  works now. 
/hold cancel

**Comment by ironcladlou** [comment]
Thanks!
/lgtm

**Pull request merged**